### PR TITLE
added AccidentNotificationBoltTest; restructured Accident and Acciden…

### LINKED ISSUE #10: Make AccidentNotificationBolt listen to two streams
Currently it listens to one stream and seperates tuples for accident from accident detection bolt and vehicle tracking by evaluating tuples. Listening to seperate streams is much more elegant.

### PR DESCRIPTION
**Pull request by krichter722**
…tImmutable (reusing code, keeping properties of PosReport in a reference to PosReport); checking 0 to 4 segments upstream for accident; corrected recording of previously notified vehicles; making AccidentNotificationBolt listen to two streams (introduced TopologyControl.ACCIDENT_INFO_STREAM_ID), closes #10

### CODE REVIEW

**Comment by krichter722** [comment]
I'll adjust the topology integration test LRBTopologyMainTest in order to deal with the handling of accident notification data on two separate streams as mentioned in #10 an it makes sense to push that into this pull request as well.

**Comment by krichter722** [comment]
All done, now, ready to merge. Sorry for the size of the PR, the alternative was to do a lot of rebasing and different PRs.

Travis builds for OpenJDK 6 seem to fail systematically. Do we need Java 6 support or can we drop it (see https://travis-ci.org/krichter722/aeolus/jobs/65496987 for details).

**Comment by mjsax** [comment]
I just merge the other two PRs. Please fix this one and rebase. If Travis is green, I will have a closer lock.

**Comment by krichter722** [comment]
As mentioned in #19 it seems to be an issue related to Hibernate that just doesn't work on Java 6. If we kick Hibernate we can replace it with a JPA provider that supports Java 6 (if any) or switch to a plain JDBC storage.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,7 @@
 target/
+queries/lrb/aeolus-test/
+queries/lrb/derby.log
+queries/lrb/nameext_lrbNormal__L2_2W_T2_1E_O0_acc
+queries/lrb/nameext_lrbNormal__L2_2W_T2_1E_O0_bal
+queries/lrb/nameext_lrbNormal__L2_2W_T2_1E_O0_exp
+queries/lrb/nameext_lrbNormal__L2_2W_T2_1E_O0_toll
```
Comment (mjsax) [inline .gitignore:9]: Was sind das für Dateien?

File: queries/lrb/src/main/java/de/hub/cs/dbis/lrb/toll/FileTollDataStore.java [hunk 2]
```diff
@@ -16,48 +16,166 @@
...
+		this.histFile = File.createTempFile("aeolus-lrb", null);
+	}
+
+	public FileTollDataStore(File histFile) throws FileNotFoundException {
 		this.histFile = histFile;
 	}
-	
-	public FileTollDataStore(String histFilePath) {
+
+	public FileTollDataStore(String histFilePath) throws FileNotFoundException {
```
Comment (mjsax) [inline queries/lrb/src/main/java/de/hub/cs/dbis/lrb/toll/FileTollDataStore.java:77]: Woher kommt diese FileNotFoundException ?
Bitte auch IllegalArgumentException documentieren.

File: queries/lrb/src/main/java/de/hub/cs/dbis/lrb/toll/MemoryTollDataStore.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	private final Map<Triple<Integer, Integer, Integer>, Integer> store = new HashMap<Triple<Integer, Integer, Integer>, Integer>();
+	private final MutableTriple<Integer, Integer, Integer> reusableMapKey = new MutableTriple<Integer, Integer, Integer>(
+		0, 0, 0);
+	
+	// avoid the allocation of memory for every key
+	
+	/**
+	 * {@inheritDoc }
+	 */
+	@Override
```
Comment (mjsax) [inline queries/lrb/src/main/java/de/hub/cs/dbis/lrb/toll/MemoryTollDataStore.java:47]: JavaDoc kann hier komplett weg gelassen werden. Die Dokumentation dann automatisch geerbt.

File: queries/lrb/src/main/java/de/hub/cs/dbis/lrb/toll/MemoryTollDataStore.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+		reusableMapKey.setLeft(xWay);
+		reusableMapKey.setMiddle(day);
+		reusableMapKey.setRight(vehicleIdentifier);
+		Integer toll = store.get(reusableMapKey);
+		return toll;
+	}
+	
+	/**
+	 * {@inheritDoc }
+	 */
```
Comment (mjsax) [inline queries/lrb/src/main/java/de/hub/cs/dbis/lrb/toll/MemoryTollDataStore.java]: JavaDoc entfernen

File: queries/lrb/src/main/java/storm/lrb/model/TollEntry.java [hunk 5]
```diff
@@ -19,22 +19,17 @@
...
-@Entity
 public class TollEntry implements Serializable {
 	private static final long serialVersionUID = 1L;
-	@Id
 	/*
-	 * internal implementation notes: - integer complies with the range of expected data in the LRB specification
-	 */
+		 * internal implementation notes: - integer complies with the range of expected data in the LRB specification
+		 */
 	private Integer vehicleIdentifier;
```
Comment (mjsax) [inline queries/lrb/src/main/java/storm/lrb/model/TollEntry.java:33]: Please correct indention.

**Pull request merged**